src/main/java/sirius/kernel/nls/NLS.java

```diff
@@ -973,16 +973,63 @@ private static String formatSpokenRecentDateWithTime(LocalDateTime givenDateTime
                   .format();
     }
 
+    /**
+     * Parses a date using the given formatter. Tries parsing with the format for the given fallback language in case
+     * of a parsing error. Falls back to the {@link #getCurrentLang() current language}, then to ISO format.
+     * <p>
+     * Throws a {@link DateTimeParseException} if the given value could not be parsed as a date.
+     *
+     * @param value                  a {@link Value} object containing the date to be parsed
+     * @param formatter              the formatter to use for parsing
+     * @param fallbackLanguageFormat the language providing a fallback date format
+     * @return the date or <tt>null</tt> if there is no source value to parse
+     * @throws DateTimeParseException if the given value could not be parsed as a date
+     */
+    @Nullable
+    public static LocalDate extractDate(@Nonnull Value value,
+                                        DateTimeFormatter formatter,
```

Comment:
Is it of interest to have a version without formatter?
And should it replicate the logic from here? https://github.com/scireum/scireum-core/blob/e3e2152855afc83a0811584adb4bff5bb256ebeb/src/main/java/scireum/core/data/CustomPropertyType.java#L583

Reply:
Since the method is now moved to `Value` the version without formatter would be `value.asLocalDate()`.

And the linked logic does not apply for this use case. During an import, a date String using a certain format is expected. It is not guaranteed that a value is given in the expected format due to user error.
So we are always in the `expectUserInput`-path of your example, but want to provide a reasonable amount of fallback options, to not reject a date if it was not given exactly in the expected format.

Reply:
OK... but than we could include more parsing to the fallback options.
NLS.parseMachineString includes the ISO case: (YYYY-MM-DD)
and NLS.parseUserString uses the current language, which could in theory replace one of your fallbacks?

Reply:
I think we should do that when needed